ambry-clustermap/src/test/java/com.github.ambry.clustermap/HelixClusterManagerTest.java

```diff
@@ -699,6 +711,53 @@ public void xidTest() throws Exception {
     assertTrue(ignoreInstanceReplica.isDown());
   }
 
+  /**
+   * Ensure that effects of the listenCrossColo config is as expected. When it is set to false, the Helix cluster manager
+   * initializes fine, but listens to subsequent InstanceConfig changes in the local colo only.
+   */
+  @Test
+  public void listenCrossColoTest() {
+    assumeTrue(!useComposite);
+    HelixClusterManager helixClusterManager = (HelixClusterManager) clusterManager;
+    Counter instanceTriggerCounter = helixClusterManager.helixClusterManagerMetrics.instanceConfigChangeTriggerCount;
+    Map<String, HelixManager> helixManagerMap = helixClusterManager.getHelixManagerMap();
+    for (Map.Entry<String, HelixManager> entry : helixManagerMap.entrySet()) {
+      if (entry.getKey().equals(localDc)) {
+        assertTrue("Helix cluster manager should always be connected to the local Helix manager",
+            entry.getValue().isConnected());
+      } else {
+        assertEquals(
+            "Helix cluster manager should be connected to the remote Helix managers if and only if listenCrossColo is"
+                + "set to false", listenCrossColo, entry.getValue().isConnected());
+      }
+    }
+    long instanceConfigChangeTriggerCount = instanceTriggerCounter.getCount();
+    helixCluster.triggerInstanceConfigChangeNotification();
+    assertEquals("Number of trigger count should be in accordance to listenCrossColo value",
+        instanceConfigChangeTriggerCount + (listenCrossColo ? dcs.length : 1), instanceTriggerCounter.getCount());
+
+    InstanceConfig remoteInstanceConfig = helixCluster.getAllInstanceConfigs()
+        .stream()
+        .filter(e -> ClusterMapUtils.getDcName(e).equals(remoteDc))
+        .findAny()
+        .get();
+    DataNodeId remote = helixClusterManager.getDataNodeId(remoteInstanceConfig.getHostName(),
+        Integer.valueOf(remoteInstanceConfig.getPort()));
+    Set<PartitionId> writablePartitions = new HashSet<>(helixClusterManager.getWritablePartitionIds(null));
+    PartitionId partitionIdToSealInRemote = helixClusterManager.getReplicaIds(remote)
+        .stream()
+        .filter(e -> writablePartitions.contains(e.getPartitionId()))
+        .findAny()
+        .get()
+        .getPartitionId();
+    remoteInstanceConfig.getRecord()
+        .setListField(SEALED_STR, Collections.singletonList(partitionIdToSealInRemote.toPathString()));
+    helixCluster.triggerInstanceConfigChangeNotification();
+    assertEquals("If replica in remote is sealed, partition should be sealed if and only if listenCrossColo is true "
+            + "and override is disabled", !listenCrossColo || overrideEnabled,
```

Comment:
If this test is only to verify listen cross colos, maybe we don't need to include `overrideEnabled` in this test, because we don't know the override states. In this test, if `overrideEnabled == true` and the override state is `ReadOnly`, the assert still may fail.  This is my two cents. Will approve.

Reply:
What the test does is mark as `ReadOnly` a partition that is guaranteed to be `ReadWrite` even if `overrideEnabled == true` - by choosing a partition on the remote that is part of `getWritablePartitions()`.